include/tuvx/linalg/linalg.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <iostream>
+#include <random>
+#include <vector>
+
+namespace tuvx {
+namespace linalg {
+
+// tridiag matrix
+template <typename T> struct trid_mat {
+  int size;
```

Comment:
```suggestion
  std::size_t size;
```
It may help to be more consistent with what the stl does, and this will prevent negative values for the size